src/components/test-selection/TestSelection.test.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from "react";
+import cases from "jest-in-case";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+import TestSelection from "./TestSelection";
+
+test("should display the test entries", () => {
+  const history = createMemoryHistory();
+
+  render(
+    <Router history={history}>
+      <TestSelection />
+    </Router>
+  );
+
+  expect(screen.getByText(/Valences Test/i)).toBeInTheDocument();
+
+  expect(screen.getByText(/ Periodic table/i)).toBeInTheDocument();
+});
+
+cases(
+  "should navigate to pages",
+  ({ itemIdx, expectedPath }) => {
+    const history = createMemoryHistory({
+      initialEntries: [
+        "/tests/valences",
+        "/tests/periodic-table",
+        "/tests/valences/settings",
+        "/tests/periodic-table/settings",
+      ],
+    });
+
+    render(
+      <Router history={history}>
+        <TestSelection />
+      </Router>
+    );
+
+    userEvent.click(screen.getAllByRole("button")[itemIdx]);
```

Comment:
The buttons within the feature are a mix of buttons and icon buttons currently there are no `name` or `aria-label` props on the buttons to help efficiently select each button. 

 This was the best approach I could think to select each button of without updating the existing button and icon button functionality.

Reply:
Awesome! And don't worry too much about not-that-nice selectors. We will be doing a followup raid converting the componenets to be functions + hooks, and we can fix up these sorts of things while we are doing that :100: